hdf5-rs/src/hl/container.rs

```diff
@@ -261,18 +277,41 @@ impl<'a> Writer<'a> {
         } else {
             let fspace = self.obj.space()?;
             let slice_shape = fspace.select_slice(slice)?;
-            let mspace = Dataspace::try_new(&slice_shape, false)?;
 
             let view = arr.into();
+            let data_shape = view.shape();
+
+            // Restore dimensions that are SliceOrIndex::Index in the slice.
+            let mut data_shape_hydrated = Vec::new();
+            let mut pos = 0;
+            for s in slice_s {
+                match s {
+                    SliceOrIndex::Index(_) => {
+                        data_shape_hydrated.push(1);
```

Comment:
1 or 0?

Reply:
1 -- SliceOrIndex::Index take a single element from that dimension.